### PR TITLE
Improve SuppressGCTransitionAttribute docs

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
Context https://github.com/dotnet/runtime/issues/39404

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SuppressGCTransitionAttribute.cs [hunk 1]
```diff
@@ -55,7 +58,12 @@ namespace System.Runtime.InteropServices
     ///   * Data corruption.
     /// </remarks>
     [AttributeUsage(AttributeTargets.Method, Inherited = false)]
-    public sealed class SuppressGCTransitionAttribute : Attribute
+#if SYSTEM_PRIVATE_CORELIB
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SuppressGCTransitionAttribute.cs:61]: This is how other similar attributes are done.

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/pull/39411 should be merged first.

File: src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj [hunk 2]
```diff
@@ -38,8 +38,8 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' != '$(NetCoreAppCurrent)'">
-    <Compile Include="$(CommonPath)System\Runtime\InteropServices\SuppressGCTransitionAttribute.internal.cs"
-             Link="Common\System\Runtime\InteropServices\SuppressGCTransitionAttribute.internal.cs" />
+    <Compile Include="$(CoreLibSharedDir)System\Runtime\InteropServices\SuppressGCTransitionAttribute.cs"
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj:41]: This won't be needed once @maryamariyan's PR is merged.

**Review by safern — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**